IdentityCore/src/cache/token/MSIDLegacyTokenCacheItem.m

```diff
@@ -215,4 +215,20 @@ - (BOOL)isTombstone
     return self.refreshToken && [self.refreshToken isEqualToString:@"<tombstone>"];
 }
 
+@synthesize oauthTokenType = _oauthTokenType;
+- (NSString *)oauthTokenType
+{
+    @synchronized (self)
```

Comment:
Maybe we can make it a readonly property so that we don't need synchronized here?
Or, just to double check, do we even need oauthTokenType property to be atomic?

Reply:
oauthTokenType is accessed by multiple threads during serialization so it has to be atomic and is set at various places so cannot be read only.

Reply:
Why not remove this code altogether and have the cacheItem return nil or a value.
Default value can be handled by the caller and that way, no need for this sync block

Reply:
As discussed, this code was needed as this property was set during decoding of the object and not during encoding which was failing the tests. Changed the code to set this property during encoding as well and removed the property override code.